src/view.c

```diff
@@ -857,51 +906,250 @@ view_on_output_destroy(struct view *view)
 	view->output = NULL;
 }
 
+#define VIEW_GET_NEXT_EDGES(_view_near_def, _view_far_def, _backward)	\
+	struct output *output = view->output;				\
+	struct server *server = output->server;				\
+	struct view *v;							\
+	wl_list_for_each(v, &server->views, link) {			\
+		if (v == view || v->output != output)			\
+			continue;					\
+		struct border margin = ssd_get_margin(v->ssd);		\
+		int vn = (_view_near_def);				\
+		int vf = (_view_far_def);				\
+		if (_backward) {					\
+			if (vf + far_gap  < point)			\
+				f = MAX(f, vf + far_gap  - point);	\
+			if (vn + near_gap < point)			\
+				n = MAX(n, vn + near_gap - point);	\
+		} else {						\
+			if (vf + far_gap  > point)			\
+				f = MIN(f, vf + far_gap  - point);	\
+			if (vn + near_gap > point)			\
+				n = MIN(n, vn + near_gap - point);	\
+		}							\
+	}
+
+
+static void
+_view_get_next_edges(struct view *view, const char *direction, int point,
+		     int max, int near_gap, int far_gap, int *near, int *far)
+{
+	/*
+	 * Returns near and far edges from all views, looking from
+	 * point towards direction. Stops searching at max.
+	 *
+	 * Note that the width and height are ignored when looking for
+	 * edges (left/right only considers x, while up/down only
+	 * considers y). This is intentional as it allows to easily
+	 * align your windows to a grid, defined by all other windows)
+	 */
+	int n = max, f = max;
+	if (!strcasecmp(direction, "left")) {
```

Comment:
I should probably use something like `enum view_edge` and `view_edge_parse()` for this, but sadly this includes `VIEW_EDGE_CENTER`. I think this should be renamed (there is no center edge), and/or combined somehow.

Reply:
> I should probably use something like `enum view_edge` and `view_edge_parse()` for this, but sadly this includes `VIEW_EDGE_CENTER`. I think this should be renamed (there is no center edge), and/or combined somehow.

While I logically agree on the `VIEW_EDGE_CENTER` part I don't see that as being an issue in practice here.
It would be the same as having a direction "foobar" in the config.

Thus, suggesting we indeed use `enum view_edge` when parsing the config so we can use a simple `switch()` here.
I am not sure if we should add a new `action_arg` type though or if we just use the existing `int` one.

Reply:
Done:
- #1007